python/instrumentation/openinference-instrumentation-langchain/src/openinference/instrumentation/langchain/_tracer.py

```diff
@@ -608,6 +598,57 @@ def _token_counts(outputs: Optional[Mapping[str, Any]]) -> Iterator[Tuple[str, i
             yield attribute_name, token_count
 
 
+def _parse_token_usage_for_non_streaming_outputs(
+    outputs: Optional[Mapping[str, Any]],
+) -> Optional[Dict[str, int]]:
+    """
+    Parses output to get token usage information for non-streaming LLMs, i.e.,
+    when `stream_usage` is set to false.
+    """
+    if (
+        outputs
+        and hasattr(outputs, "get")
+        and (llm_output := outputs.get("llm_output"))
+        and hasattr(llm_output, "get")
+        and (
+            token_usage := _get_first_value(
+                llm_output,
+                (
+                    "token_usage",
+                    "usage",  # Anthropic-specific key
+                ),
+            )
+        )
+    ):
+        return token_usage
+    return None
+
+
+def _parse_token_usage_for_streaming_outputs(
+    outputs: Optional[Mapping[str, Any]],
+) -> Optional[Dict[str, int]]:
+    """
+    Parses output to get token usage information for streaming LLMs, i.e., when
+    `stream_usage` is set to true.
+    """
+    if (
+        outputs
+        and hasattr(outputs, "get")
+        and (generations := outputs.get("generations"))
+        and hasattr(generations, "__getitem__")
+        and hasattr(generations[0], "__getitem__")
+        and (generation := generations[0][0])
```

Comment:
index could be out of bound?
```suggestion
        and hasattr(generations, "__getitem__")
        and generations
        and hasattr(generations[0], "__getitem__")
        and generations[0]
        and (generation := generations[0][0])
```

Reply:
Good catch [da68fd7](https://github.com/Arize-ai/openinference/pull/669/commits/da68fd7157fb045ab1ab82d7939d08a9b76baac8)